modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/tree/BPlusTree.java

```diff
@@ -424,6 +457,11 @@ else if (needBackIfRouting) {
             if (needWal)
                 wal.log(new ReplaceRecord<>(grpId, pageId, io, newRowBytes, idx));
 
+            if (reconciliationCtx != null &&
```

Comment:
It might be executed several times. But thanks to additional checks in the reconciliation logic, this did not lead to incorrect calculation of the size.
Moved to "if (lvl == 0) {" section. It seems that now it will not be executed several times.